expr.go

```diff
@@ -1993,6 +1993,28 @@ func evalExpr(e *expr, from, until int32, values map[metricRequest][]*metricData
 			return percentile(values, percent, interpolate)
 		})
 
+	case "removeBelowValue":
+		args, err := getSeriesArg(e.args[0], from, until, values)
+		if err != nil {
+			return nil
+		}
+		threshold, err := getFloatArg(e, 1)
+		if err != nil {
+			return nil
+		}
+
+		var results []*metricData
+		for _, metric := range args {
+			for i, value := range metric.Values {
+				if value < threshold {
+					metric.Values[i] = math.NaN()
```

Comment:
The issue for mutating the original API values is that if you're updating the metric in the values[] map, then any other part of the original target expression that uses that metric will also get the mutated values.

For example, in
`/render/?target=sum(foo.bar.baz,removeBelowValue(foo.bar.baz,20)`
carbonapi queries carbonzipper once for the `foo.bar.baz` metric, and the values for that get stuffed into the values map which is passed to `evalExpr()`.  Mutated values here will produce the wrong answer.